Add blog post for entry.runtime_data (#2160)
home-assistant/developers.home-assistant: blog/2024-04-30-store-runtime-data-inside-config-entry.md

---
author: Marc Mueller
authorURL: https://github.com/cdce8p
title: "Store runtime data inside the config entry"
---

> Often integration need to setup and keep track of custom data like coordinators, api connections or code objects. Previously those would all be stored inside `hass.data` which made keeping track of it difficult.
> 
> With recent changes, it's now possible to use `entry.runtime_data` for that. The config entry is already available when setting up platforms and gets cleaned up automatically. No more deleting the key from `hass.data` after unload.
> 
> It also better supports type checking. `ConfigEntry` is generic now, so it's possible to pass the data type along. Use a typed data structure like `dataclass` for that. To simplify the annotation, it's recommended to define a type alias for it.

Reviewer: ```suggestion
Integrations often need to set up and track custom data, such as coordinators, API connections, or code objects. Previously, those were all stored inside `hass.data,` which made tracking them difficult.

With recent changes, it's now possible to use `entry.runtime_data` for that. The config entry is already available when setting up platforms and gets cleaned up automatically. No more deleting the key from `hass.data` after unloading.

It also better supports type-checking. `ConfigEntry` is generic now, so passing the data type along is possible. Use a typed data structure like `dataclass` for that. To simplify the annotation, it's recommended to define a type alias for it.
```

> An example could look like this

Reviewer: ```suggestion
An example could look like this:

```
